pkg/serverless/protocol.go

```diff
@@ -130,12 +142,8 @@ func (d *Daemon) TriggerFlush(ctx context.Context, shutdown bool) {
 	// we've just flushed, we can maybe try to change the flush strategy?
 	// (but do that only if the flush strategy hasn't be forced through configuration)
 	// note that we don't mind doing that if we are shutting down.
-	if d.useAdaptiveFlush && !shutdown {
-		newStrat := d.AutoSelectStrategy()
-		if newStrat.String() != d.flushStrategy.String() {
-			log.Debug("Switching to flush strategy:", newStrat)
-			d.flushStrategy = newStrat
-		}
+	if !shutdown {
```

Comment:
Looks like the comment above this might need to be updated (I already changed it in my PR though so this could happen as part of a merge conflict resolution)